sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SessionNotAvailableRetryTest.java

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import com.azure.cosmos.BridgeInternal;
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.implementation.directconnectivity.ConsistencyReader;
+import com.azure.cosmos.implementation.directconnectivity.ConsistencyWriter;
+import com.azure.cosmos.implementation.directconnectivity.ReflectionUtils;
+import com.azure.cosmos.implementation.directconnectivity.ReplicatedResourceClient;
+import com.azure.cosmos.implementation.directconnectivity.ServerStoreModel;
+import com.azure.cosmos.implementation.directconnectivity.StoreClient;
+import com.azure.cosmos.implementation.directconnectivity.StoreReader;
+import com.azure.cosmos.implementation.directconnectivity.StoreResponse;
+import com.azure.cosmos.implementation.directconnectivity.TransportClient;
+import com.azure.cosmos.implementation.directconnectivity.Uri;
+import com.azure.cosmos.models.CosmosItemRequestOptions;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.cosmos.rx.TestSuiteBase;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.mockito.Mockito;
+import org.mockito.stubbing.Answer;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.mockito.Mockito.doAnswer;
+
+public class SessionNotAvailableRetryTest extends TestSuiteBase {
+    private CosmosAsyncClient client;
+    private CosmosAsyncContainer cosmosAsyncContainer;
+    private DatabaseAccount databaseAccount;
+
+    @BeforeClass(groups = {"simple"}, timeOut = SETUP_TIMEOUT)
+    public void beforeClass() throws IllegalAccessException {
+        client = new CosmosClientBuilder()
```

Comment:
resource leak. this is not closed anywhere.

Reply:
done